### ISSUE TITLE
enum_variant_names: false positive on part of a longer word

### ISSUE THREAD

**Issue by dtolnay** [rust-lang-nursery/rust-clippy#1241]
``` rust
pub enum Pat {
    /// Represents a wildcard pattern (`_`)
    Wild,

    /// A struct or struct variant pattern, e.g. `Variant {x, y, ..}`.
    /// The `bool` is `true` in the presence of a `..`.
    Struct(Path, Vec<FieldPat>, bool),

    /// A tuple struct/variant pattern `Variant(x, y, .., z)`.
    /// If the `..` pattern fragment is present, then `Option<usize>` denotes its position.
    /// 0 <= position <= subpats.len()
    TupleStruct(Path, Vec<Pat>, Option<usize>),

    /// A possibly qualified path pattern.
    /// Unquailfied path patterns `A::B::C` can legally refer to variants, structs, constants
    /// or associated constants. Quailfied path patterns `<A>::B::C`/`<A as Trait>::B::C` can
    /// only legally refer to associated constants.
    Path(Option<QSelf>, Path),
}
```

```
warning: Variant name starts with the enum's name, #[warn(enum_variant_names)] on by default
   --> src/expr.rs:262:5
    |
262 |     Path(Option<QSelf>, Path),
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^
```

"Path" does not start with "Pat" (i.e. "Pattern").

**Comment by dtolnay** [comment]
I also hit this with pub_enum_variant_names.
